src/main/kotlin/org/rust/lang/core/completion/RustCompletionEngine.kt

```diff
@@ -28,7 +32,15 @@ object RustCompletionEngine {
         }
 
         val visitor = CompletionScopeVisitor(ref)
-        for (scope in enumerateScopesFor(ref)) {
+        val scopes = enumerateScopesFor(ref).toMutableList()
+
+        val rootMod = RustResolveUtil.getCrateRootModFor(ref)
+        if (rootMod != null) {
+            val preludeMod = rootMod.module?.preludeModule
+            if (preludeMod != null && preludeMod is RustFile) scopes += preludeMod
+        }
```

Comment:
I think that you can simply do `ref.module?.preludeModule` without resolving the crate root. When we implement proper handing of `#[no_std]` attributes and the like we will certainly need to consult with the crate root file. For now it's better not to guess how that API will look like and instead use the simplest solution.